bin/compile

```diff
@@ -1,5 +1,12 @@
 #!/usr/bin/env bash
 
+function log() {
+        echo "-----> $1"
+        if [ -n "$2" ]; then
+                exit $2
+        fi
```

Comment:
```suggestion
```

Not a fan of this. I wouldn't expect a logging function to also exit with an exit code. Better keep the exits like before.

Reply:
I'm undecided myself - have seen this pattern often in shell scripts, but that's not necessarily is a good thing - @marns93 / @mm-matthias do you have an opinion on this?

Reply:
I am with @stohrendorf and would also go for separating logging and exit code logic. Maybe have a 2nd function like log_and_exit that combines log + exit if one really wants to combine this?

Since this is a shell script my standards are fairly low in any case 😄

Reply:
> Since this is a shell script my standards are fairly low in any case

...which leads most people to combining log and exit.